pkg/operation/common/utils.go

```diff
@@ -106,50 +107,39 @@ func GenerateAddonConfig(values map[string]interface{}, enabled bool) map[string
 }
 
 // DeleteHvpa delete all resources required for the HVPA in the given namespace.
-func DeleteHvpa(ctx context.Context, k8sClient kubernetes.Interface, namespace string) error {
-	if k8sClient == nil {
-		return fmt.Errorf("require kubernetes client")
-	}
-
-	listOptions := metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("%s=%s", v1beta1constants.GardenRole, v1beta1constants.GardenRoleHvpa),
-	}
+func DeleteHvpa(ctx context.Context, c client.Client, namespace string) error {
+	labelSelectorHVPA := client.MatchingLabels{v1beta1constants.GardenRole: v1beta1constants.GardenRoleHvpa}
 
 	// Delete all CRDs with label "gardener.cloud/role=hvpa"
 	// Workaround: Due to https://github.com/gardener/gardener/issues/2257, we first list the HVPA CRDs and then remove
 	// them one by one.
-	crdList, err := k8sClient.APIExtension().ApiextensionsV1beta1().CustomResourceDefinitions().List(ctx, listOptions)
-	if err != nil {
+	crdList := &apiextensionsv1beta1.CustomResourceDefinitionList{}
```

Comment:
You could use `v1` now that the min Kubernetes version for seeds was raised to `1.18`.

Reply:
Yep, done.
I also bumped the [technical seed version requirement](https://github.com/gardener/gardener/blob/4e518bb440525fb2adc5796d70c4b0003c2c5f02/pkg/operation/seed/seed.go#L1395) to 1.18 in order to reflect the doc change in https://github.com/gardener/gardener/pull/4426.